[tayjerom] iP

tayjerom opened the pull request:
Creation of PR - Week 4

tayjerom committed:
Level-0

tayjerom committed:
Level-1: Echo

tayjerom committed:
Level-2: Add, List

tayjerom committed:
Level-3

tayjerom committed:
A-CodingStandard

tayjerom committed:
[Project De-Cluttering]

Extracted repetitive code into separate methods

tayjerom committed:
Level 4. Create Classes: [ToDo], [Event], [Deadlines] - inherit [Task] Class

AaronZZ10 commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,177 @@
+        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+    }
+
+    private static void addTask(String taskDescription) {
```
maybe add some comments to explain the function of the method and its parameters

AaronZZ10 commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,177 @@
+        try {
+            int taskIndex = Integer.parseInt(commands[1]) - 1;
+            if (isValidTaskIndex(taskIndex)) {
+                tasks[taskIndex].markAsDone();
```
maybe define a method to print those line to provide better readability

AaronZZ10 commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,177 @@
+        if (taskCount < MAX_TASKS) {
+
+            String[] parts = taskDescription.split(" ", 2);
+            String categoryOfTask = parts[0];
```
i think "category" should be enough and effient

AaronZZ10 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        isDone = true;
+    }
+
+    public void removeAsDone() {
```
I find the naming of "removeAsDone" quite confusing

AaronZZ10 reviewed (commented):
the code looks pretty good and clear overall, good job!

tayjerom committed:
[A-CodeQuality]:

i) Changed task types: declared as CONSTANTS: TODO, DEADLINE, and EVENT
ii) Refactored addTask: into [createTask], [createDeadlineTask], and [createEventTask].
iii) Updated [handleMarkCommand] to handle error cases and unnecessary nesting removed.

tayjerom committed:
Implement Level-5: Handle Errors

tayjerom committed:
Merge branch 'branch-Level-5'

tayjerom committed:
Implement A-Exceptions: handle errors with custom exceptions

tayjerom committed:
Merge branch 'branch-A-Exceptions'

tayjerom committed:
A-Packages: organize classes into packages under task

tayjerom committed:
Merge branch 'branch-A-Packages'

tayjerom committed:
Implement Level 6: Delete feature into branch-Level-6.

tayjerom committed:
Merge branch 'branch-Level-6'

tayjerom committed:
Update to A-Collections extension: ArrayList<Task> to store task

tayjerom committed:
Merge branch 'branch-Level-7' into branch-Level-6

tayjerom committed:
branch-Level-6: fix errors in the delete functions.

tayjerom committed:
Merge branch 'branch-Level-6'

tayjerom committed:
Implement Level 7:Saving tasks in hard disk

tayjerom committed:
Implement Level 7: Saving tasks into hard drive

tayjerom committed:
Added user manual for better user experience

tzerbin commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,243 @@
+import task.UserManual;
+
+public class JerChatBot {
+    private static final String FILE_PATH = "./data/data.txt";
```
Nice usage of constants!

tzerbin commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,243 @@
+
+public class JerChatBot {
+    private static final String FILE_PATH = "./data/data.txt";
+    private static final ArrayList<Task> tasks = new ArrayList<>();
```
Is there a reason why tasks and datahandling are set as final?
Constants should also be in SCREAMING_SNAKE_CASE.

tzerbin commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,243 @@
+public class JerChatBot {
+    private static final String FILE_PATH = "./data/data.txt";
+    private static final ArrayList<Task> tasks = new ArrayList<>();
+    private static final FileHandling datahandling = new FileHandling(FILE_PATH);
```
If datahandling is not a constant, then it should be named as dataHandler.
Verbs are reserved for methods.

tzerbin commented on src/main/java/JerChatBot.java:
```diff
@@ -0,0 +1,243 @@
+
+            try {
+                switch (command) {
+                case "bye":
```
Perhaps there is more to be done for magic strings and integers for the entire codebase?

tzerbin commented on src/main/java/datahandling/FileHandling.java:
```diff
@@ -0,0 +1,133 @@
+
+public class FileHandling {
+
+    private final String filePath;
```
Constants should be initialized in the beginning, not during runtime.

tzerbin commented on src/main/java/datahandling/FileHandling.java:
```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileHandling {
```
As mentioned previously, class names should be nouns.

tzerbin commented on src/main/java/exception/InvalidTaskNumberException.java:
```diff
@@ -0,0 +1,7 @@
+package exception;
+
+public class InvalidTaskNumberException extends Exception {
```
Very nice postfix "Exception" for exception classes!

tzerbin commented on src/main/java/task/ToDo.java:
```diff
@@ -0,0 +1,13 @@
+package task;
+
+public class ToDo extends Task{
```
Perhaps there is a better way to name this class?

tzerbin reviewed (commented):
Good attempt on following the guidelines!

There is a number of naming issues for constants and classes, together with the usage of magic strings / numbers.

tayjerom committed:
Refactored JerChatBot to align with OOP principles.

tayjerom committed:
Implement Level-8: Dates and Times

tayjerom committed:
Implement Level-9: Find task

tayjerom committed:
Merge pull request #1 from tayjerom/branch-Level-8

Implement Level-8: Dates and Times

tayjerom committed:
Merge pull request #2 from tayjerom/branch-Level-9

Implement Task Search function (Level-9)

tayjerom committed:
Resolve code changes after PR merge (Level 8/9) and Update Test Script for (find) command.

tayjerom committed:
Added JavaDoc Comments

tayjerom committed:
Merge pull request #3 from tayjerom/branch-A-JavaDoc

Added A-JavaDoc comments

tayjerom committed:
Update README.md file (A-UserGuide)

Update User Guide for JerChatBot on README.md file (A-UserGuide)

tayjerom committed:
Update README file with JerChatBot User Guide

tayjerom committed:
Merge pull request #4 from tayjerom/branch-A-UserGuide

Update README file (A-UserGuide)

tayjerom committed:
Remove test files and junk files
